src/app/dashboard/game.service.ts

```diff
@@ -0,0 +1,160 @@
+import { Injectable } from '@angular/core';
+import Konva from 'konva';
+import { BehaviorSubject } from 'rxjs';
+import { Hero } from '../core/interface/hero';
+
+const COLOR_HEALTH = {
+  SAFE: '#198754',
+  DANGER: '#dc3545',
+  BACKGROUND: '#ccc'
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GameService {
+  heroesPlaying = new BehaviorSubject<Hero[]>([]);
+  updateGame = new BehaviorSubject<boolean>(false);
+  drawAgain = false;
+
+  constructor() { }
+
+  updateHeroesPlaying(hero: Hero[], drawAgain?: boolean) {
+    this.drawAgain = !!drawAgain;
+    this.heroesPlaying.next(hero);
+  }
+
+  /**
+   * Draw a ring
+   * @param x: x position
+   * @param y: y position
+   * @param radius: radius of outer ring
+   * @param color: color ring
+   * @returns Konva Ring object
+   */
+  drawRing(x: number, y: number, radius: number, color: string): Konva.Ring {
+    return new Konva.Ring({
+      x,
+      y,
+      innerRadius: radius - 6,
+      outerRadius: radius,
+      fill: color,
+      strokeWidth: 0
+    })
+  }
+
+  /**
+   * draw a konva arc
+   * @param x 
+   * @param y 
+   * @param radius 
+   * @param color 
+   * @param angle 
+   * @param id 
+   * @returns 
+   */
+  drawArc(x: number, y: number, radius: number, color: string, angle: number, id: string) {
+    return new Konva.Arc({
+      x,
+      y,
+      innerRadius: radius - 6,
+      outerRadius: radius,
+      angle,
+      fill: color,
+      strokeWidth: 0,
+      rotation: -90,
+      clockwise: true,
+      id
+    });
+  }
+
+  /**
+   * get angle of arc
+   * @param number 
+   * @param total 
+   * @returns 
+   */
+  getAngle(number: number, total: number, ratio: number): number {
+    return number / total * 360 * ratio;
+  }
+
+  /**
+   * draw an image
+   * @param imageSrc 
+   * @param config
+   * @param group 
+   * @param layer 
+   */
+  drawImage(imageSrc: string, config: {x: number, y: number, width: number, height: number}, group: Konva.Group, layer: Konva.Layer) {
+    Konva.Image.fromURL(imageSrc, (image: any) => {
+      image.setAttrs(config);
+      group.add(image);
+      layer.batchDraw();
+    });
+  }
```

Comment:
exception handling?

Reply:
This is a void function for draw image. Input is required with their type so I not do exception handling